### PR TITLE
libclipboard: init at 1.1

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/libclipboard/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  buildInputs = [ libxcb libXau libXdmcp ];
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" ];
+  outputs = [ "out" "dev" ];
+
+  meta = {
+    description = "Lightweight cross-platform clipboard library";
```
Comment (Aleksanaa) [inline pkgs/by-name/li/libclipboard/package.nix:29]: still `homepage`

**Review by Aleksanaa — COMMENTED** [review]
This looks a bit dead. Do we need it for other packages?

**Comment by Sigmanificient** [comment]
> This looks a bit dead. Do we need it for other packages?

Despite not receiving much updates, it's a library, so it nice always a plus to provide it.

Originally i wanted to package `lyssa` (C++ music player) which requires it, but i couldn't manage to make a working derivation for it, i made a issue for it: #310636.

**Comment by Aleksanaa** [comment]
```
/nix/store/g0gn91m56b267ncx05w93kihyqia39cm-builder.sh: line 82: code: No space left on device
```

@ofborg build libclipboard

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 311224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libclipboard</li>
    <li>libclipboard.dev</li>
  </ul>
</details>

**Comment by Aleksanaa** [comment]
```
/tmp/nix-build-libclipboard.drv-0/source/src/clipboard_cocoa.c:17:10: fatal error: 'Cocoa/Cocoa.h' file not found
#include <Cocoa/Cocoa.h>
         ^~~~~~~~~~~~~~~
1 error generated.
```

@wegank could you help with this 🥹 Looks like `darwin.apple_sdk.frameworks.Foundation`

File: pkgs/by-name/li/libclipboard/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  version = "1.1";
+
+  src = fetchFromGitHub {
+    owner = "jtanx";
+    repo = "libclipboard";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-553hNG8QUlt/Aff9EKYr6w279ELr+2MX7nh1SKIklhA=";
+  };
+
+  buildInputs = [ libxcb libXau libXdmcp ];
```
Comment (wegank) [inline pkgs/by-name/li/libclipboard/package.nix:23]:
```suggestion
  buildInputs = [ libxcb libXau libXdmcp ]
    ++ lib.optional stdenv.isDarwin [ darwin.apple_sdk.frameworks.Cocoa ];
```

Comment (Sigmanificient) [reply pkgs/by-name/li/libclipboard/package.nix:23]: let me try this!

**Pull request merged**